frontend/src/js/query-runner/DownloadResultsDropdownButton.tsx

```diff
@@ -43,56 +44,76 @@ const Separator = styled("div")`
 // might need to adjust this when adding more content.
 const dropdownOffset: [number, number] = [-37, 8]; // [skidding, distance] / default [0, 10]
 
-const getEnding = (url: string) => url.split(".").reverse()[0].toUpperCase();
+interface FileChoice {
+  label: string;
+  ending: string;
+}
 
-const getInitialEndingChoice = (resultUrls: string[]) => {
-  const { preferredDownloadFormat } = getUserSettings();
+const getEnding = (url: string) => url.split(".").reverse()[0].toUpperCase();
 
+const getInitialEndingChoice = (resultUrls: ResultUrlsWithLabel[]) => {
+  const { preferredDownloadFormat, preferredDownloadLabel } = getUserSettings();
+  const labelResult = resultUrls.find(
+    (url) => url.label === preferredDownloadLabel,
+  );
+  if (labelResult) {
+    return labelResult;
+  }
   const found = resultUrls.find(
-    (url) => getEnding(url) === preferredDownloadFormat,
+    (url) => getEnding(url.url) === preferredDownloadFormat,
   );
 
-  return found ? getEnding(found) : getEnding(resultUrls[0]);
+  return found ? found : resultUrls[0];
 };
 
 const DownloadResultsDropdownButton = ({
   resultUrls,
   tiny,
   tooltip,
 }: {
-  resultUrls: string[];
+  resultUrls: ResultUrlsWithLabel[];
   tiny?: boolean;
   tooltip?: string;
 }) => {
-  const [endingChoice, setEndingChoice] = useState(
-    getInitialEndingChoice(resultUrls),
-  );
+  const [fileChoice, setFileChoice] = useState<FileChoice>(() => {
+    const initial = getInitialEndingChoice(resultUrls);
+    return { label: initial.label, ending: getEnding(initial.url) };
+  });
 
   useEffect(() => {
-    storeUserSettings({ preferredDownloadFormat: endingChoice });
-  }, [endingChoice]);
-
-  const urlChoice = useMemo(
-    () =>
-      resultUrls.find((url) => getEnding(url) === endingChoice) ||
-      resultUrls[0],
-    [resultUrls, endingChoice],
-  );
+    storeUserSettings({
+      preferredDownloadFormat: fileChoice.ending,
+      preferredDownloadLabel: fileChoice.label,
+    });
+  }, [fileChoice]);
+
+  const urlChoice = useMemo(() => {
+    const labelResult = resultUrls.find(
+      (url) => url.label === fileChoice.label,
+    );
+    if (labelResult) {
+      return labelResult;
+    }
+    return (
+      resultUrls.find((url) => getEnding(url.url) === fileChoice.ending) ||
+      resultUrls[0]
+    );
+  }, [resultUrls, fileChoice]);
 
   const dropdown = useMemo(() => {
     return (
       <List>
         {resultUrls.map((url) => {
```

Comment:
If you destructure like this you can avoid url.url
```suggestion
        {resultUrls.map(({ url, label }) => {
```